src/main/kotlin/blackjack/domain/BlackjackResults.kt

```diff
@@ -0,0 +1,23 @@
+package blackjack.domain
+
+class BlackjackResults(participants: Participants) {
+    val dealerResult: DealerResult
+    val gamblerResults: List<GamblerResult>
+
+    init {
+        val dealer = participants.extractDealer()
+        val gamblers = participants.extractGamblers()
+
+        gamblerResults = determineGamblerResults(dealer, gamblers)
+        dealerResult = DealerResult(dealer, gamblerResults)
+    }
+
+    private fun determineGamblerResults(
+        dealer: Dealer,
+        gamblers: List<Gambler>,
+    ): List<GamblerResult> {
+        return gamblers.map { gambler ->
+            GamblerResult(gambler, gambler.determineResultStatus(dealer))
+        }
```

Comment:
아래와 같이 gambler 객체에게 GamblerResult를 결정해달라고 메시지를 보내보면 어떨까요? 🤔
```suggestion
        return gamblers.map { it.determinResult(dealer) }
```

Reply:
반영 해보았는데, 이 편이 훨씬 깔끔하네요!!